site/cds_rdm/inspire_harvester/transform_entry.py

```diff
@@ -146,69 +184,251 @@ def _transform_publication_date(self):
-                f"PIRE record id: {self.inspire_metadata.get('control_number')}."
-            )
-            return None
+        for author in authors:
+            inspire_roles = author.get("inspire_roles", [])
+            if "supervisor" in inspire_roles:
+                contributors.append(author)
 
-        result_doc_type = document_type_mapping.get(document_type)
-        return result_doc_type
+        return self._transform_creatibutors(contributors) + mapped_corporate_authors
 
     def _transform_creators(self):
-        """Mapping of INSPIRE authors to creators and contributors."""
+        """Mapping of INSPIRE authors to creators."""
+        authors = self.inspire_metadata.get("authors", [])
         creators = []
-        authors = self.inspire_metadata.get("authors")
+        for author in authors:
+            inspire_roles = author.get("inspire_roles")
+            if not inspire_roles:
+                creators.append(author)
+            elif "author" in inspire_roles or "editor" in inspire_roles:
+                creators.append(author)
+
+        return self._transform_creatibutors(creators)
+
+    def _transform_creatibutors(self, authors):
+        """Transform creatibutors."""
+        creatibutors = []
         try:
             for author in authors:
                 first_name = author.get("first_name")
                 last_name = author.get("last_name")
 
-                rdm_creator = {
+                rdm_creatibutor = {
                     "person_or_org": {
                         "type": "personal",
                     }
                 }
 
                 if first_name:
-                    rdm_creator["person_or_org"]["given_name"] = first_name
+                    rdm_creatibutor["person_or_org"]["given_name"] = first_name
                 if last_name:
-                    rdm_creator["person_or_org"]["family_name"] = last_name
+                    rdm_creatibutor["person_or_org"]["family_name"] = last_name
                 if first_name and last_name:
-                    rdm_creator["person_or_org"]["name"] = (
+                    rdm_creatibutor["person_or_org"]["name"] = (
                         author.get("last_name") + ", " + author.get("first_name")
                     )
-                creators.append(rdm_creator)
-            return creators
+
+                creator_affiliations = self._transform_author_affiliations(author)
+                creator_identifiers = self._transform_author_identifiers(author)
+                role = author.get("inspire_roles")
+
+                if creator_affiliations:
+                    rdm_creatibutor["affiliations"] = creator_affiliations
+
+                if creator_identifiers:
+                    rdm_creatibutor["person_or_org"][
+                        "identifiers"
+                    ] = creator_identifiers
+
+                if role:
+                    rdm_creatibutor["role"] = role[0]
```

Comment:
is it possible that one author has multiple roles? we could add them multiple times to the entry with different role each time, no?

Reply:
Discussed irl, decided to take the 1st item